apps/src/StudioApp.js

```diff
@@ -1842,6 +1843,13 @@ StudioApp.prototype.clearAndAttachRuntimeAnnotations = function() {
  * Click the reset button.  Reset the application.
  */
 StudioApp.prototype.resetButtonClick = function() {
+  if (!this.hasReported) {
+    this.report({
+      app: getStore().getState().pageConstants.appType,
+      level: this.config.level.id
+    });
+  }
+  this.hasReported = false;
```

Comment:
should this be the last line of the report method? so we don't have to remember to reset it outside of the `report` function?

Reply:
is the idea here that if we already reported for some other reason, you want to wait until the next button click to report again? if so i think it might be clearer to rename the property to something like `shouldReportResetButtonClick`, and set to `true` here and `false in `report`.

Reply:
>is the idea here that if we already reported for some other reason, you want to wait until the next button click to report again?

Yes and no. I'd put it more like this: We want to ensure we report a milestone exactly once per run/reset cycle. We know clicking the reset button indicates a new cycle has started, so if we haven't reported a milestone by this point, this is our last chance and we need to do it here. 
It ends up being dependent by app. Some apps always report a milestone when the run button is clicked. Others always report a milestone when the validation logic completes. Others never report a milestone during the run/reset cycle. This allows apps to keep reporting milestones when they normally would but adds in a milestone report for the apps that never report.

Reply:
> should this be the last line of the report method? so we don't have to remember to reset it outside of the report function?

Due to the "once-per-cycle" goal, we will need to reset it somewhere that is guaranteed to happen once. The `resetButtonClick` function seemed like an appropriate place because it "resets" the state of the app.

Reply:
For the variable name. What do you think of something like `hasReportedMilestone` or `hasReportedMilestoneThisCycle`

Reply:
Given you explanation above i think `hasReported` is perfectly fine, and the locations of `=true` and `=false` make sense. But you could put that explanation in a comment?